File: LeetCode/1790/1790.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        auto view = std::views::zip(s1,s2) | std::views::filter([&](const auto& par){
            auto [left, right] = par;
            return left != right;
        });
        std::vector<pair<char,char>> vec(view.begin(), view.end());
        switch (vec.size())
        {
            case 0:
                return true;
                break;
            case 2:
                return vec[0].first == vec[1].second && vec[0].second == vec[1].first;
                break;
            default: return false;

        }
        return false;
    }
};

File: LeetCode/1790/1790.rs
impl Solution {
    pub fn are_almost_equal(s1: String, s2: String) -> bool 
    {
        let indexies: Vec<_> = s1.chars().zip(s2.chars()).enumerate().filter(|(_, (left, right))| left != right).collect();

        match indexies.len()
        {
            0 => true,
            2 => {
                let (_, (a1, b1)) = indexies[0];
                let (_, (a2,b2)) = indexies[1];
                a1 == b2 && a2 == b1
            
            },
            _ => false
        }

    }
}
